pkg/controller/config-controller.go

```diff
@@ -0,0 +1,313 @@
+	verbose                                        string // verbose levels: 1, 2, ...
+	uploadProxyServiceName                         string
+}
+
+//NewConfigController creates a new ConfigController
+func NewConfigController(client kubernetes.Interface,
+	cdiClientSet clientset.Interface,
+	ingressInformer extensioninformers.IngressInformer,
+	routeInformer routeinformers.RouteInformer,
+	configInformer informers.CDIConfigInformer,
+	uploadProxyServiceName string,
+	pullPolicy string,
+	verbose string) *ConfigController {
+	c := &ConfigController{
+		client:                 client,
+		cdiClientSet:           cdiClientSet,
+		queue:                  workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
+		ingressInformer:        ingressInformer.Informer(),
+		routeInformer:          routeInformer.Informer(),
+		configInformer:         configInformer.Informer(),
+		ingressLister:          ingressInformer.Lister(),
+		routeLister:            routeInformer.Lister(),
+		configLister:           configInformer.Lister(),
+		ingressesSynced:        ingressInformer.Informer().HasSynced,
+		routesSynced:           routeInformer.Informer().HasSynced,
+		configsSynced:          configInformer.Informer().HasSynced,
+		uploadProxyServiceName: uploadProxyServiceName,
+		pullPolicy:             pullPolicy,
+		verbose:                verbose,
+	}
+
+	// Bind the ingress SharedIndexInformer to the ingress queue
+	c.ingressInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: c.handleObjAdd,
+		UpdateFunc: func(old, new interface{}) {
+			newDepl := new.(*extensionsv1beta1.Ingress)
+			oldDepl := old.(*extensionsv1beta1.Ingress)
+			if newDepl.ResourceVersion == oldDepl.ResourceVersion {
+				// Periodic resync will send update events for all known PVCs.
+				// Two different versions of the same PVCs will always have different RVs.
+				return
+			}
+			c.handleObjUpdate(new)
+		},
+
+		DeleteFunc: c.handleObjDelete,
+	})
+
+	// Bind the route SharedIndexInformer to the route queue
+	c.routeInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: c.handleObjAdd,
+		UpdateFunc: func(old, new interface{}) {
+			newDepl := new.(*routev1.Route)
+			oldDepl := old.(*routev1.Route)
+			if newDepl.ResourceVersion == oldDepl.ResourceVersion {
+				// Periodic resync will send update events for all known PVCs.
+				// Two different versions of the same PVCs will always have different RVs.
+				return
+			}
+			c.handleObjUpdate(new)
+		},
+
+		DeleteFunc: c.handleObjDelete,
+	})
+
+	return c
+}
+
+func (c *ConfigController) handleObjAdd(obj interface{}) {
+	c.handleObject(obj, "add")
+}
+func (c *ConfigController) handleObjUpdate(obj interface{}) {
+	c.handleObject(obj, "update")
+}
+func (c *ConfigController) handleObjDelete(obj interface{}) {
+	c.handleObject(obj, "delete")
+}
+
+func (c *ConfigController) handleObject(obj interface{}, verb string) {
```

Comment:
These callback functions really just decide whether the Route/Ingress/whatever is relevant to us (refers to our service) and if so, enqueue something to be processed a worker thread.  Everything else is ignored.

I don't actually see any queuing happening anywhere in this file so `processNextWorkItem()` must never be getting called